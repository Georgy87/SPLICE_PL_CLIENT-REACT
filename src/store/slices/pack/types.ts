import { Samples } from "../samples/types";

export type PacksSliceState = {
    packs: Pack[] | [];
    packProfile: null | Pack;
    userPacks: Pack[] | [];
    loading: boolean;
}

export type createPackType = {
	info: {
		genre: string;
		authorName: string;
		packInfo: string;
	};
	picture: File;
	audio: File;
};

export type Pack = {
    _id: string;
    genre: string;
    name: string;
    packInfo: string;
    text: string;
    listens: number;
    picture: string;
    audio: string;
    pause: boolean;
    samples: Samples[];
    userId: string;
}


