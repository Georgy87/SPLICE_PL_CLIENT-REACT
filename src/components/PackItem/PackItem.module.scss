%download {
	display: flex;
	justify-content: center;
	align-items: center;
	height: 140px;
	width: 250px;
	border: 1px solid #ddddd9;
	margin: 0 auto;
}

.packCardWrapper {
	display: flex;
	width: 99%;
	.packCard {
		position: relative;
		display: flex;
		flex-direction: column;
		width: 17rem;
		img {
			border-radius: 7px;
			height: 17rem;
		}
		.playPauseCircle {
			display: flex;
			position: absolute;
			width: 100%;
			height: calc(100% - 40px);
			align-items: center;
			justify-content: center;
			button {
				background-color: transparent;
				border: none;
			}
		}
	}
}

.packCardWrapper.changePage {
	display: block;
	.packCard {
		.playPauseCircle {
			display: none;
		}
	}
}

.drop {
	@extend %download;
}

.downloadSamples {
	@extend %download;
	margin-top: 30px;
}
