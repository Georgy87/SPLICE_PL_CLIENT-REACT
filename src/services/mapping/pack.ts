import {
  CreatePackResponseDto,
  GetPackResponseDto,
  GetPacksResponseDto,
  GetUserPacksResponseDto,
  SearchPacksResponseDto,
} from '@services/api/types';

import {
  CreatePackResponseData,
  GetPackResponseData,
  GetPacksResponseData,
  GetUserPacksResponseData,
  SearchPacksResponseData,
} from './types';

export const packCreatePackResponseDto = (dto: CreatePackResponseData): CreatePackResponseDto => {
  return dto.data;
};

export const packGetPacksResponseDto = (dto: GetPacksResponseData): GetPacksResponseDto => {
  return dto.data;
};

export const packGetPackResponseDto = (dto: GetPackResponseData): GetPackResponseDto => {
  return dto.data;
};

export const packGetUserPacksResponseDto = (dto: GetUserPacksResponseData): GetUserPacksResponseDto => {
  return dto.data;
};

export const packSearchPacksResponseDto = (dto: SearchPacksResponseData): SearchPacksResponseDto => {
  return dto.data;
};
