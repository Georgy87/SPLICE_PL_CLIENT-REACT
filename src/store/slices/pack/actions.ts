import { createAsyncThunk } from '@reduxjs/toolkit';

import {
  CreatePackResponseDto,
  GetPacksResponseDto,
  GetUserPacksResponseDto,
  SearchPacksResponseDto,
} from '@api/types';
import { packsApi } from '@api/packsApi';

import { createPackType } from './types';

export const fetchCreatePack = createAsyncThunk('packs/createPackStatus', async (payload: createPackType) => {
  try {
    const { picture, audio } = payload;
    const { genre, authorName, packInfo } = payload.info;

    const formData = new FormData();
    formData.append('genre', genre);
    formData.append('name', authorName);
    formData.append('packInfo', packInfo);
    formData.append('picture', picture);
    formData.append('audio', audio);

    const data: CreatePackResponseDto = await packsApi.createPack(formData).promise;
    return data;
  } catch (error) {
    console.log(error);
  }
});

export const fetchGetPacks = createAsyncThunk('packs/getPacksStatus', async (payload: number) => {
  try {
    const data: GetPacksResponseDto = await packsApi.getPacks(payload).promise;
    return data;
  } catch (error) {
    console.log(error);
  }
});

export const fetchGetPack = createAsyncThunk(
  'packs/getPackStatus',
  async (payload: { packId: string; tag: string | null }) => {
    try {
      return packsApi.getPack(payload).promise;
    } catch (error) {
      console.log(error);
    }
  }
);

export const fetchGetUserPacks = createAsyncThunk('packs/getUserPacksStatus', async () => {
  try {
    const data: GetUserPacksResponseDto = await packsApi.getUserPacks().promise;
    return data;
  } catch (error) {
    console.log(error);
  }
});

export const fetchSearchPacks = createAsyncThunk('packs/getSearchPacksStatus', async (payload: string) => {
  try {
    const data: SearchPacksResponseDto = await packsApi.searchPacks(payload).promise;
    return data;
  } catch (error) {
    console.log(error);
  }
});
