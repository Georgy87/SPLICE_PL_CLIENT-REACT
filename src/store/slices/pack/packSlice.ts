import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';

import { packsApi } from '../../../services/api/packsApi';
import { createPackType, Pack, PacksSliceState } from './types';

const initialState: PacksSliceState = {
	packs: [],
	packProfile: null,
	userPacks: [],
};

export const fetchCreatePack = createAsyncThunk(
	'packs/createPackStatus',
	async (payload: createPackType) => {
		try {
			const { picture, audio } = payload;
			const { genre, authorName, packInfo } = payload.info;

			const formData = new FormData();
			formData.append('genre', genre);
			formData.append('name', authorName);
			formData.append('packInfo', packInfo);
			formData.append('picture', picture);
			formData.append('audio', audio);

			const packs = await packsApi.createPack(formData);
			return packs;
		} catch (error) {
			console.log(error);
		}
	},
);

export const fetchGetPacks = createAsyncThunk('packs/getPacksStatus', async () => {
	try {
		const packs = await packsApi.getPacks();
		return packs;
	} catch (error) {
		console.log(error);
	}
});

export const fetchGetPack = createAsyncThunk('packs/getPackStatus', async (packId: string) => {
	try {
		const pack = await packsApi.getPack(packId);
		return pack;
	} catch (error) {
		console.log(error);
	}
});

export const fetchGetUserPacks = createAsyncThunk('packs/getUserPacksStatus', async () => {
	try {
		const packs = await packsApi.getUserPacks();
		return packs;
	} catch (error) {
		console.log(error);
	}
});

export const fetchSearchPacks = createAsyncThunk(
	'packs/getSearchPacksStatus',
	async (search: string) => {
		try {
			const pack = await packsApi.searchPacks(search);
			return pack;
		} catch (error) {
			console.log(error);
		}
	},
);

export const packSlice = createSlice({
	name: 'packs',
	initialState,
	reducers: {
		setDefaultPackState: (state) => {
			state.packs = [];
			state.packProfile = null;
			state.userPacks = [];
		},
	},
	extraReducers: (builder) =>
		builder
			.addCase(fetchGetPacks.fulfilled.type, (state, action: PayloadAction<Pack[]>) => {
				state.packs = action.payload;
			})
			.addCase(fetchCreatePack.fulfilled.type, (state, action: PayloadAction<Pack[]>) => {
				state.packs = action.payload;
			})
			.addCase(fetchGetPack.fulfilled.type, (state, action: PayloadAction<Pack>) => {
				state.packProfile = action.payload;
			})
			.addCase(fetchGetUserPacks.fulfilled.type, (state, action: PayloadAction<Pack[]>) => {
				state.userPacks = action.payload;
			})
			.addCase(fetchSearchPacks.fulfilled.type, (state, action: PayloadAction<Pack[]>) => {
				state.packs = action.payload;
			}),

});

export const { setDefaultPackState } = packSlice.actions;

export const packsReducer = packSlice.reducer;
