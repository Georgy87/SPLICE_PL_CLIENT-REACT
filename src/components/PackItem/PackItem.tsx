import React, { useEffect, useState } from 'react';
import { useHistory } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';

import { useActions } from '../../hooks/useAction';
import { Pack } from '../../store/types/packs';
import { useTypedSelector } from '../../hooks/useTypedSelector';
import { selectPacks } from '../../store/selectors/packsSelectors';
import { selectAudio } from '../../store/selectors/playerSelectors';
import { pausePack, playPack } from '../../store/slices/packSlice';
import { IconChangeLayout } from '../../layouts/IconChangeLayout/IconChangeLayout';

import styles from './PackItem.module.scss';

type PackListProps = {
	pack: Pack;
	active?: boolean;
	id: string;
};

export const PackItem: React.FC<PackListProps> = ({ pack, id }) => {
	const [packId, setPackId] = useState<string>('');

	const audio = useSelector(selectAudio);

	const { active, pause, volume } = useTypedSelector((state) => state.player);

	const dispatch = useDispatch();

	const router = useHistory();

	const {
		playTrack,
		pauseTrack,
		setActiveTrack,
		setAudioPlay,
		setAudioPause,
		setAudioSrc,
	} = useActions();

	const play = async (id: string) => {
		if (pack.pause) {
			setActiveTrack({ pack, flag: true });
			await setAudioSrc('http://localhost:5000/' + active?.audio);

			dispatch(playPack(id));
			setAudioPlay();
			playTrack();
		} else {
			dispatch(pausePack(id));
			setAudioPause();
			pauseTrack();
		}
	};

	return (
		<div className={styles.packCard}>
			<IconChangeLayout
				onClicked={() => play(pack._id)}
				blockStyle={styles.playPauseCircle}
				iconOneOrTwo={pack.pause}
				iconOne='play'
				iconTwo='pause'
				iconStyle={{
					color: '#fff',
					fontSize: '60px',
					cursor: 'pointer',
				}}
			>
			</IconChangeLayout>
			<img src={`http://localhost:5000/${pack.picture}`} />

			<div>
				<div>{pack.name}</div>
				<div style={{ fontSize: 12, color: 'gray' }}>{pack.artist}</div>
			</div>

			{/* {packId === active?._id && <div>02:42 / 03:22</div>} */}
		</div>
	);
};
