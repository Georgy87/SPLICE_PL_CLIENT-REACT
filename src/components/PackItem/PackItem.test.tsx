import { render } from '@testing-library/react';
import { pack } from '../../store/slices/pack/testData';
import { renderWithRouter } from '../../tests/helpers/RenderWithRouter';
import { PackItem } from './PackItem';

describe('PackItem component', () => {
	test('PackItem snapshot', async () => {
		const packItem = render(
			renderWithRouter(<PackItem pack={pack} id='621fe5b9815ea94e0e103a89' index={0} />),
		);
		expect(packItem).toMatchSnapshot();
	});
});
