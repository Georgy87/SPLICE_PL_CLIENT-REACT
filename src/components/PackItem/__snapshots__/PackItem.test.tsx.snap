// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PackItem component PackItem snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="root"
        data-testid="packs-page"
      >
        <div
          class="player"
        >
          <video
            autoplay=""
            class="playerVideo"
            loop=""
            src="https://storage.yandexcloud.net/sample-cloud/videos/ARCADE%20by%20Output.mp4"
          />
          <div
            class="initialInformation"
          >
            <h1>
              More than a sample library
            </h1>
            <p>
              Samples are a source of inspiration. It’s what you do with them that matters. SampleCloud allows you to shape millions of royalty-free samples to your sound.
            </p>
          </div>
        </div>
        <div
          class="inputContainer"
        >
          <label>
            Sounds
          </label>
          <div
            class="search"
          >
            <svg
              class="searchIcon"
              viewBox="0 0 512 512"
              width="95"
            >
              <path
                d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
              />
            </svg>
            <input
              class="searchInput"
              placeholder=""
              type="text"
              value=""
            />
            <svg
              class="searchClose"
              viewBox="0 0 352 512"
              width="100"
            >
              <path
                d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
              />
            </svg>
            <svg
              class="searchDelete"
              viewBox="0 0 640 512"
              width="100"
            >
              <path
                d="M576 64H205.26A63.97 63.97 0 0 0 160 82.75L9.37 233.37c-12.5 12.5-12.5 32.76 0 45.25L160 429.25c12 12 28.28 18.75 45.25 18.75H576c35.35 0 64-28.65 64-64V128c0-35.35-28.65-64-64-64zm-84.69 254.06c6.25 6.25 6.25 16.38 0 22.63l-22.62 22.62c-6.25 6.25-16.38 6.25-22.63 0L384 301.25l-62.06 62.06c-6.25 6.25-16.38 6.25-22.63 0l-22.62-22.62c-6.25-6.25-6.25-16.38 0-22.63L338.75 256l-62.06-62.06c-6.25-6.25-6.25-16.38 0-22.63l22.62-22.62c6.25-6.25 16.38-6.25 22.63 0L384 210.75l62.06-62.06c6.25-6.25 16.38-6.25 22.63 0l22.62 22.62c6.25 6.25 6.25 16.38 0 22.63L429.25 256l62.06 62.06z"
              />
            </svg>
          </div>
        </div>
        <div
          class="root"
        >
          <div
            class="loaderWrapper"
          >
            <div
              class="loader"
            >
              <div
                class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
                role="progressbar"
                style="width: 40px; height: 40px;"
              >
                <svg
                  class="MuiCircularProgress-svg"
                  viewBox="22 22 44 44"
                >
                  <circle
                    class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                    cx="44"
                    cy="44"
                    fill="none"
                    r="20.2"
                    stroke-width="3.6"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="loaderPagination"
          />
        </div>
      </div>
      <div
        class="packCardWrapper"
      >
        <a
          class="packCard"
          data-testid="profile-pack-link"
          href="/profile-pack/621fe5b9815ea94e0e103a89"
        >
          <div
            class="playPauseCircle"
          >
            <div
              class="platformBtn"
            >
              <button
                class="platformBtn"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  style="color: rgb(255, 255, 255); font-size: 60px; cursor: pointer;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 14.5v-9l6 4.5-6 4.5z"
                  />
                </svg>
              </button>
            </div>
          </div>
          <img
            alt="pack-cover"
            src="https://sample-cloud.storage.yandexcloud.net/PACK-IMAGES/5dd58548-f621-4fad-b985-3a03d22dd574.jpeg"
          />
          <div>
            <div>
              Bass House
            </div>
            <div
              style="font-size: 12px; color: gray;"
            >
              James Blake
            </div>
          </div>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="root"
      data-testid="packs-page"
    >
      <div
        class="player"
      >
        <video
          autoplay=""
          class="playerVideo"
          loop=""
          src="https://storage.yandexcloud.net/sample-cloud/videos/ARCADE%20by%20Output.mp4"
        />
        <div
          class="initialInformation"
        >
          <h1>
            More than a sample library
          </h1>
          <p>
            Samples are a source of inspiration. It’s what you do with them that matters. SampleCloud allows you to shape millions of royalty-free samples to your sound.
          </p>
        </div>
      </div>
      <div
        class="inputContainer"
      >
        <label>
          Sounds
        </label>
        <div
          class="search"
        >
          <svg
            class="searchIcon"
            viewBox="0 0 512 512"
            width="95"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            />
          </svg>
          <input
            class="searchInput"
            placeholder=""
            type="text"
            value=""
          />
          <svg
            class="searchClose"
            viewBox="0 0 352 512"
            width="100"
          >
            <path
              d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
            />
          </svg>
          <svg
            class="searchDelete"
            viewBox="0 0 640 512"
            width="100"
          >
            <path
              d="M576 64H205.26A63.97 63.97 0 0 0 160 82.75L9.37 233.37c-12.5 12.5-12.5 32.76 0 45.25L160 429.25c12 12 28.28 18.75 45.25 18.75H576c35.35 0 64-28.65 64-64V128c0-35.35-28.65-64-64-64zm-84.69 254.06c6.25 6.25 6.25 16.38 0 22.63l-22.62 22.62c-6.25 6.25-16.38 6.25-22.63 0L384 301.25l-62.06 62.06c-6.25 6.25-16.38 6.25-22.63 0l-22.62-22.62c-6.25-6.25-6.25-16.38 0-22.63L338.75 256l-62.06-62.06c-6.25-6.25-6.25-16.38 0-22.63l22.62-22.62c6.25-6.25 16.38-6.25 22.63 0L384 210.75l62.06-62.06c6.25-6.25 16.38-6.25 22.63 0l22.62 22.62c6.25 6.25 6.25 16.38 0 22.63L429.25 256l62.06 62.06z"
            />
          </svg>
        </div>
      </div>
      <div
        class="root"
      >
        <div
          class="loaderWrapper"
        >
          <div
            class="loader"
          >
            <div
              class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
              role="progressbar"
              style="width: 40px; height: 40px;"
            >
              <svg
                class="MuiCircularProgress-svg"
                viewBox="22 22 44 44"
              >
                <circle
                  class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                  cx="44"
                  cy="44"
                  fill="none"
                  r="20.2"
                  stroke-width="3.6"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="loaderPagination"
        />
      </div>
    </div>
    <div
      class="packCardWrapper"
    >
      <a
        class="packCard"
        data-testid="profile-pack-link"
        href="/profile-pack/621fe5b9815ea94e0e103a89"
      >
        <div
          class="playPauseCircle"
        >
          <div
            class="platformBtn"
          >
            <button
              class="platformBtn"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                style="color: rgb(255, 255, 255); font-size: 60px; cursor: pointer;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 14.5v-9l6 4.5-6 4.5z"
                />
              </svg>
            </button>
          </div>
        </div>
        <img
          alt="pack-cover"
          src="https://sample-cloud.storage.yandexcloud.net/PACK-IMAGES/5dd58548-f621-4fad-b985-3a03d22dd574.jpeg"
        />
        <div>
          <div>
            Bass House
          </div>
          <div
            style="font-size: 12px; color: gray;"
          >
            James Blake
          </div>
        </div>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
